docs/proposals/templates-v2.md

```diff
@@ -0,0 +1,334 @@
+It's the slower but more powerful version of `Foam: Create New Note`, allowing you to pick any template, as well as customize he filepath.
+
+## Open questions
+
+### Arbitrary hotkey -> template mappings?
+
+`Foam: Open Daily Note` and `Foam: Create New Note` only differ by their hotkey and their default template setting.
+Is there a reason/opportunity to abstract this further and allow for users to define custom `hotkey -> template` mappings?
+
+```jsonc
+"foam.templateHotkeys": {
+  "default": {
+    "hotkey": "Alt+C"
+    "template": "my_default_template.md"
+  },
+  "daily": {
+    "hotkey": "Alt+D",
+    "template": "my_daily_template.md"
+  },
+  "Fancy Note": {
+    "hotkey": "Alt+E",
+    "template": "my_fancy_note_template.md"
+  }
+}
+```
+
+## Limitations of this proposal
+
+### No `dateFormat`
+
+`foam.openDailyNote.filenameFormat` uses `dateFormat()` to put the current timestamp into the daily notes filename. This is much more flexible than what is available in VSCode Snippet variables. Before daily notes are switched over to use templates, we might have to come up with another mechanism/syntax to allow for calls to `dateFormat()` within template files.
+
+This would be especially useful in the migration of users to the new daily notes templates. For example, if `foam.defaultDailyNoteTemplate` is unset, then we could generate an implicit template for use by `Foam: Open Daily Note`. Very roughly something like:
+
+```markdown
+<!-- The below front-matter block is for foam-specific template settings -->
+---
+<!-- The default filepath to use when using this template -->
+<!-- Relative paths are relative to the workspace, absolute paths are absolute -->
+<!-- Note that you can include VSCode snippet variables to template the path -->
+filepath: `#{foam.openDailyNote.directory}/${foam.openDailyNote.filenameFormat}.${foam.openDailyNote.fileExtension}`
+---
+
+<!-- The actual contents of the template begin after the `---` thematic break immediately below this line-->
+---
+# ${foam.openDailyNote.titleFormat}
+```
+
+## Extensions
+
+In addition to the ideas of this proposal, there are ways we could imagine extending it. These are all "out of scope" for this design, but thinking about them could be useful to guide our thinking about this design.
+
+#### More variables in templates
+
+`${title}` is necessary in this case to replace the functionality of `Markdown Notes: New Note`.
+However, one could imagine that this pattern of "Ask the user for a value for missing variable values" could be useful in other situations too.
+Perhaps users could even define their own (namespaced) template variables, and Foam would ask them for values to use for each when creating a note using a template that used those variables.
+
+#### Schedule-based templating
+
+As suggested by @ingalless [here](https://github.com/foambubble/foam/issues/523#issuecomment-799231399), it might be handy if different templates could be used depending on the day of the week.
```

Comment:
Ah - sorry! I think my example drew unnecessary confusion. You can remove this here, or summarise the below if you like.

Let's say I've configured the `foam.defaultDailyNoteTemplate` configuration. I'd probably set this to something like `work-day.md` because 5 days a week are working. For the weekend though I'd probably want a bail out - something like "Create daily note from template" or "Create blank daily note" so that I don't generate my default template.

In fact, your proposal already nicely covers this - it's just a couple extra commands around daily notes.

Reply:
> In fact, your proposal already nicely covers this

Glad to hear it. I've removed that section.